sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsGdsStore.scala

```diff
@@ -102,17 +104,18 @@ class RapidsGdsStore(
   }
 
   private def singleShotSpill(other: RapidsBuffer, deviceBuffer: DeviceMemoryBuffer)
-      : RapidsBufferBase = {
+  : RapidsBufferBase = {
     val id = other.id
     val path = id.getDiskPath(diskBlockManager)
+    val alignedSize = RapidsGdsStore.alignBufferSize(deviceBuffer)
```

Comment:
It would be nice to have comments explaining why it's desired and safe to deliberately walk off the known end of the buffer, otherwise every occurrence of `alignBufferSize` looks broken and pointless.

As an aside, I can see the desire to align the address of both the src and dest, but is the length really that important when we're not appending/sharing a file?

Reply:
Ok, added some comments about the alignment approach.